userspace/libsinsp/utils.h

```diff
@@ -133,6 +134,12 @@ class sinsp_utils
         // timezone of Z for UTC, but does support parsing fractional seconds,
         // unlike get_epoch_utc_seconds_* below.
 	static bool parse_iso_8601_utc_string(const std::string& time_str, uint64_t &ns);
+
+	template<typename Duration = std::chrono::seconds>
+	static uint32_t get_steady_clock_ts()
```

Comment:
`uint32_t` for a Unix time? 2038 is closer than it seems ;)

Is it possible to switch this to a higher resolution (e.g. nanoseconds like with `get_current_time_ns()` as well? Either way, please add a unit to the method name (`_sec`, `_ns` or whatever you end up with)

Reply:
the method is templatized by Duration, having seconds as a default duration. It does not have any sense to add a suffix to the method name to indicate the unit (the prefix is the template specialisation). Maybe I could add the clock type as a new template parameter to have also the high-resolution clock.

Yes, I could use  `get_current_time_ns`, even if I think that using `chrono` library make the code more portable, and moreover I want to have a steady clock instead of a system clock here.
Anyway, this is not so important, so ping me if you want to switch to `get_current_time_ns` or whatever change you want

Reply:
I' ve split the method in two. `get_duration_from_epoch` which is templatized by clock type and tick period, and `get_monotonic_ts_sec` which specialize the method by steady_clock and seconds

Reply:
I feel it's overcomplicating things, `get_current_time_ns()` seems simple enough and we can survive with wall clock time until we switch to monotonic